packages/admin/cms-admin/src/generator/future/generateForm.ts

```diff
@@ -28,6 +28,15 @@ export function generateForm(
     const numberFields = config.fields.filter((field) => field.type == "number");
     const booleanFields = config.fields.filter((field) => field.type == "boolean");
 
+    const isFieldOptional = (fieldName: string): boolean => {
```

Comment:
we also have `FieldConfig.required` (that falls back to introspected value it not defined)

- either we support `required` here too
- or we drop required and use introspection only

in any case, your helper function should be used in `generateFormField` too, I'd move it into `utils/`

Reply:
done 👍